app/misc.py

```diff
@@ -153,6 +153,10 @@ def __init__(self, userclass=None, subs=(), prefs=()):
         elif config.site.allow_uploads and (config.site.upload_min_level <= get_user_level(self.uid, self.score)[0]):
             self.canupload = True
 
+        self.can_pm_users = False
+        if config.site.send_pm_to_user_min_level <= get_user_level(self.uid, self.score)[0] or self.admin:
+            self.can_pm_users = True
+
```

Comment:
This should probably be a function, otherwise we're running `get_user_level` on every request made by a logged in user when we don't even need that data. Also, `SiteAnon` should have the same function returning always `False`.